lstchain/image/modifier.py

```diff
@@ -439,119 +510,167 @@ def calculate_required_additional_nsb(simtel_filename, data_dl1_filename, config
+    # We also get the tel_id to which the DL1 file corresponds:
+    median_ped_meanpixq, median_ped_stdpixq, tel_id = get_pix_median_charges(
+            data_dl1_filename, EventType.SKY_PEDESTAL)
 
+    # Now we process the Monte Carlo:
     # Event reader for simtel file:
     mc_reader = EventSource(input_url=simtel_filename, config=Config(config))
 
-    # Obtain the configuration with which the pedestal calculations were
-    # performed:
-    ped_config = config['LSTCalibrationCalculator']['PedestalIntegrator']
-    tel_id = ped_config['tel_id']
-    # Obtain the (unbiased) extractor used for pedestal calculations:
-    pedestal_calibrator = CameraCalibrator(
-        image_extractor_type=ped_config['charge_product'],
-        config=Config(config['LSTCalibrationCalculator']),
-        subarray=mc_reader.subarray)
-
-    # Since these extractors are now for use on MC, we have to apply the pulse
-    # integration correction (in data that is currently, as of
-    # lstchain v0.7.5, replaced by an empirical (hard-coded) correction of the
+    subarray = mc_reader.subarray
+
+    # Get the single-pe response fluctuations:
+    spe_location = (config['waveform_nsb_tuning']['spe_location']
+                    if 'spe_location' in config['waveform_nsb_tuning']
+                       and config['waveform_nsb_tuning']['spe_location']
+                       is not None
+                    else get_resource_path(
+            "data/SinglePhE_ResponseInPhE_expo2Gaus.dat"))
+    spe = np.loadtxt(spe_location).T
+    spe_integral = np.cumsum(spe[1])
+    charge_spe_cumulative_pdf = interp1d(spe_integral, spe[0], kind='cubic',
+                                         bounds_error=False, fill_value=0.,
+                                         assume_sorted=True)
+    # Pulse template shape for a single p.e.:
+    pulse_templates = {tel_id: NormalizedPulseTemplate.load_from_eventsource(
+            subarray.tel[tel_id].camera.readout, resample=True) for tel_id in
+        config['source_config']['LSTEventSource']['allowed_tels']}
+
+    # Since the pulse integrator will now be used on MC, we have to apply the
+    # pulse integration correction (in data that is currently, as of
+    # lstchain v0.10, replaced by an empirical (hard-coded) correction of the
     # adc to pe conversion factors )
-    pedestal_calibrator.image_extractors[ped_config['charge_product']].apply_integration_correction = True
-
-    # MC pedestals integrated with the unbiased pedestal extractor
-    mc_ped_charges = []
+    config['LocalPeakWindowSum']['apply_integration_correction'] = True
+    r1_dl1_calibrator = CameraCalibrator(image_extractor_type=config[
+        'image_extractor'], config=Config(config), subarray=subarray)
+
+    numevents = 0
+    maxmcevents = 200 # Enough statistics to determine the right NSB level
+
+    # Simulated levels of additional NSB, rate in p.e./ns (a.k.a. GHz):
+    total_added_nsb = np.array([0, 0.125, 0.25, 0.5, 1, 2, 4])
```

Comment:
No, it is "total" as opposed to "incremental" because the added noise accumulates on the waveforms (which is good, it saves computing time here). It does not refer to "all the levels" we are testing.